main.go

```diff
@@ -1,150 +1,126 @@
-	dg, err = discordgo.New("Bot " + conf.Token)
+	l, err = logging.New()
 	if err != nil {
-		errorLog.Println("Error creating Discord session,", err)
-		return
+		log.Fatalf("Failed to create bot's logger: %s", err)
 	}
+	defer l.Close()
 
-	// Open a websocket connection to Discord and begin listening.
-	err = dg.Open()
+	dg, err = discordgo.New("Bot " + conf.Token)
 	if err != nil {
-		errorLog.Println("Error opening connection,", err)
+		l.Errorf("Failed to create Discord session: %s", err)
 		return
 	}
-	defer dg.Close()
-
-	//Register handlers
 	dg.AddHandler(messageCreate)
-	//dg.AddHandler(membJoin)
 
-	setInitialGame(dg)
-
-	fmt.Fprintln(logF, "")
-	infoLog.Println(`/*********BOT RESTARTED*********\`)
-
-	// Wait here until CTRL-C or other term signal is received.
-	fmt.Println("Bot is now running. Press CTRL-C to exit.")
+	if err := dg.Open(); err != nil {
+		l.Errorf("Failed to open websocket connection: %s", err)
+		return
+	}
+	defer dg.Close()
 
 	http.HandleFunc("/help", help)
-
-	errorLog.Fatalln(http.ListenAndServe(":4201", nil))
+	if err := http.ListenAndServe(":4201", nil); err != nil {
+		l.Errorf("Failed to serve HTTP: %s", err)
+	}
 }
 
 func help(w http.ResponseWriter, r *http.Request) {
-	resp := helpBody{}
+	resp := &helpBody{}
 
 	bytes, err := ioutil.ReadAll(r.Body)
 	if err != nil {
-		dg.ChannelMessageSend("354748497683283979", err.Error())
+		l.Errorf("Failed to read request body: %s", err)
+		dg.ChannelMessageSend(conf.HelpChannelId, "help request error, check logs")
 		return
 	}
-
-	err = json.Unmarshal(bytes, &resp)
+	r.Body.Close()
+	err = json.Unmarshal(bytes, resp)
 	if err != nil {
-		dg.ChannelMessageSend("354748497683283979", err.Error())
+		l.Errorf("Failed to unmarshal request JSON %q: %s", bytes, err)
+		dg.ChannelMessageSend(conf.HelpChannelId, "help request error, check logs")
 		return
 	}
-
-	dg.ChannelMessageSend("354748497683283979", fmt.Sprintf("```From: %s\nEmail: %s\n\nSubject: %s\n\n%s```", resp.User, resp.Email, resp.Subject, resp.Message))
+	msg := fmt.Sprintf("```From: %s\nEmail: %s\n\nSubject: %s\n\n%s```", resp.User, resp.Email, resp.Subject, resp.Message)
+	dg.ChannelMessageSend(conf.HelpChannelId, msg)
 }
 
+// messageCreate is an event handler which is called whenever a new message
+// is created in the Discord server.
 func messageCreate(s *discordgo.Session, m *discordgo.MessageCreate) {
-	if m.Author.Bot || !strings.HasPrefix(m.Content, conf.Prefix) || strings.TrimPrefix(m.Content, conf.Prefix) == ""{
+	if m.Author.Bot || !strings.HasPrefix(m.Content, conf.Prefix) ||
+		strings.TrimPrefix(m.Content, conf.Prefix) == "" {
 		return
 	}
-
-	parseCommand(s, m, strings.TrimPrefix(m.Content, conf.Prefix))
-}
-
-func loadLog() *os.File {
-	var err error
-	logF, err = os.OpenFile("log.log", os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644)
-	if err != nil {
-		fmt.Println(err)
-		os.Exit(2)
-	}
-	return logF
-}
-
-func setInitialGame(s *discordgo.Session) {
```

Comment:
game status gone? i use it to show the prefix of the bot in case people forget/for new people

Reply:
I will add this back in, I couldn't see what it was actually doing :P